core-java/src/main/java/org/spine3/server/RepositoryBase.java

```diff
@@ -47,16 +49,29 @@ protected RepositoryBase() {
     }
 
     private Constructor<E> getEntityConstructor() {
-        final Constructor<E> result;
+        final Class<E> entityClass = getEntityClass();
+        final Class<I> idClass = getIdClass();
         try {
-            final Class<E> entityClass = getEntityClass();
-            final Class<I> idClass = getIdClass();
+            final Constructor<E> result = entityClass.getDeclaredConstructor(idClass);
+            checkConstructorAccessModifier(result.getModifiers(), entityClass.getSimpleName());
```

Comment:
We need the name of the class only when modifiers are not right. But we get it from the class all the time.
Why cannot we get the class name right before we generate the exception message?
Please also note my comment on using FQN for the entity class name below.